metashare/api/models.py

```diff
@@ -658,6 +667,27 @@ def remove_scratch_org(self, error):
         # SCRATCH_ORG_DELETE event:
         self.delete(should_finalize=False)
 
+    def queue_submit_review(self, *, user, data):
+        from .jobs import submit_review_job
+
+        submit_review_job.delay(user=user, scratch_org=self, data=data)
+
+    def finalize_submit_review(self, timestamp, err=None, delete_org=False):
+        """
+        This could be on the task, as it relates only to updating the
+        task, but that would make a strange and hard-to-follow asymmetry
+        with the queue_submit_review method above.
+        """
+        self.task.review_submitted_at = timestamp
+        if err:
+            self.task.review_valid = False
+        else:
+            self.task.review_valid = True
+            if delete_org:
+                self.queue_delete()
+        self.task.save()
+        self.task.notify_changed()
```

Comment:
@wlonk Should we also add a `task.currently_submitting_review` boolean flag? It would fit the pattern... I think practically it would only be helpful if the user refreshes the page during review submission, in preventing them from re-submitting again.

Reply:
Yeah, I think it'd be useful.